商品購入機能

jodyyang007 opened the pull request:
# What 
商品購入機能
# Why
商品購入できるように実装

jodyyang007 committed:
商品名など購入画面に表示する

jodyyang007 committed:
購入入力フォームの保存の実装

jodyyang007 committed:
単体テストコード実装

jodyyang007 committed:
クレジットカードの入力index実装

jodyyang007 committed:
購入したら、Token及びDBに情報を保存され

jodyyang007 committed:
テスト実装

jodyyang007 committed:
sold outの実装

jodyyang007 committed:
sold outしたら、購入ボタン表示しません

jodyyang007 commented:
- [ ]  入力に問題がある状態で「購入」ボタンが押された場合、情報は受け入れられず、購入ページでエラーメッセージが表示される動画
https://gyazo.com/347d0ad07e7093cb81406c1c63e16b40

- [ ] 必要な情報を適切に入力して「購入」ボタンを押すと、商品の購入ができる動画
https://gyazo.com/9bb17280004a2a00879058da04701dd2
https://gyazo.com/9f714476882c067dfdc9ab0a461223b0

- [ ] 売却済みの商品は、画像上に「sold out」の文字が表示される動画（商品一覧機能実装時に未実装であった場合）
- [ ] 売却済みの商品は、画像上に「sold out」の文字が表示される動画（商品詳細機能実装時に未実装であった場合）
- [ ] ログイン状態の場合でも、売却済みの商品には、「商品の編集」「削除」「購入画面に進む」ボタンが表示されない動画（商品詳細機能実装時に未実装であった場合）
https://gyazo.com/3beb8aa58063a49a2796fee4e67c6df2

- [ ] テスト結果の画像
https://gyazo.com/fb29d24eb19da5857736fb090992b661

jodyyang007 commented:
- [ ] ログアウト状態の場合は、URLを直接入力して商品購入ページに遷移しようとすると、商品の販売状況に関わらずログインページに遷移する動画
https://gyazo.com/2b7394d5e15daf9e2c6ec62628e31e2d

jodyyang007 committed:
ログアウト状態の場合は、URLを直接入力して商品購入ページに遷移しようとすると、商品の販売状況に関わらずログインページに遷移する動画

jodyyang007 commented:
- [ ] ログイン状態の場合でも、URLを直接入力して自身が出品していない売却済み商品の商品購入ページへ遷移しようとすると、トップページに遷移する動画
https://gyazo.com/a7a072d3078c58dfca51a85300696237
- [ ]  ログイン状態の場合でも、URLを直接入力して自身が出品した商品の商品購入ページに遷移しようとすると、商品の販売状況に関わらずトップページに遷移する動画
https://gyazo.com/7d9e3ca8e3b9cdefca64cd610e9b3b63

- [ ]  ログイン状態の場合でも、URLを直接入力して自身が出品した売却済み商品の商品情報編集ページへ遷移しようとすると、トップページに遷移する動画（商品情報編集機能実装時に未実装であった場合）
https://gyazo.com/6580fa544910f1bda0b3e8975f88ec4f

- [ ]  ログアウト状態の場合は、URLを直接入力して商品購入ページに遷移しようとすると、商品の販売状況に関わらずログインページに遷移する動画
https://gyazo.com/2109715db11306ff91ac029829bcfb57

jodyyang007 committed:
ログイン状態の場合でも、URLを直接入力して自身が出品した売却済み商品の商品情報編集ページへ遷移しようとすると、トップページに遷移

jodyyang007 committed:
ログイン状態の場合でも、URLを直接入力して自身が出品していない売却済み商品の商品購入ページへ遷移しようとすると、トップページに遷移

ログイン状態の場合でも、URLを直接入力して自身が出品した商品の商品購入ページに遷移しようとすると、商品の販売状況に関わらずトップページに遷移

ushi10 commented:
レビュー依頼ありがとうございます。
それぞれ動画を確認いたしました。
一部動画において挙動が確実に確認できなかったため改めて動画のご提出をお願いいたします。

>  ログイン状態の場合でも、URLを直接入力して自身が出品していない売却済み商品の商品購入ページへ遷移しようとすると、トップページに遷移する動画
> https://gyazo.com/a7a072d3078c58dfca51a85300696237
> 
>  ログイン状態の場合でも、URLを直接入力して自身が出品した商品の商品購入ページに遷移しようとすると、商品の販売状況に関わらずトップページに遷移する動画
> https://gyazo.com/7d9e3ca8e3b9cdefca64cd610e9b3b63
> 
>  ログイン状態の場合でも、URLを直接入力して自身が出品した売却済み商品の商品情報編集ページへ遷移しようとすると、トップページに遷移する動画（商品情報編集機能実装時に未実装であった場合）
> https://gyazo.com/6580fa544910f1bda0b3e8975f88ec4f


上記それぞれ「出品者」が確認できませんでした。
ログインユーザー、出品者（詳細ページに表示されております）をそれぞれ映したのちに画面遷移をするような動画を撮影しましょう。

撮影完了後、こちらのプルリクエストに動画を添付していただきレビュー依頼フォームのご提出をよろしくお願いいたします。

jodyyang007 committed:
修正１

jodyyang007 commented:
ログイン状態の場合でも、URLを直接入力して自身が出品していない売却済み商品の商品購入ページへ遷移しようとすると、トップページに遷移する動画
https://gyazo.com/f4b033f820620cc1758f9de3e1e6b77f
https://gyazo.com/0e0cd970211feb95600a8c6873bb4950

ログイン状態の場合でも、URLを直接入力して自身が出品した商品の商品購入ページに遷移しようとすると、商品の販売状況に関わらずトップページに遷移する動画
https://gyazo.com/9405d7708ca4f4d850bde1338cad2112
https://gyazo.com/739cc58a2b2d8a622c973cd7c6637850

ログイン状態の場合でも、URLを直接入力して自身が出品した売却済み商品の商品情報編集ページへ遷移しようとすると、トップページに遷移する動画（商品情報編集機能実装時に未実装であった場合）
https://gyazo.com/09cc996ab328ee3454a5c9903375ca36

minachanchan reviewed (commented):
レビューのご依頼ありがとうございます！

以下について正確に挙動の確認が出来ませんでしたので、大変恐れ入りますが改めてキャプチャの撮影をお願いできますでしょうか？

```
・ログイン状態の場合でも、URLを直接入力して自身が出品していない売却済み商品の商品購入ページへ遷移しようとすると、トップページに遷移する動画
ログインユーザー名と商品詳細ページの出品者名、soldoutの表記がわかるよう状態から購入画面への直接遷移の様子を撮影いただけますと幸いです。


・ログイン状態の場合でも、URLを直接入力して自身が出品した売却済み商品の商品情報編集ページへ遷移しようとすると、トップページに遷移する動画（商品情報編集機能実装時に未実装であった場合）
自身が出品した売却済み商品の「編集ページ」への遷移について撮影いただけますと幸いです。
```

Gyazoで撮影いただいたキャプチャのURLをこちらのプルリクエストに添付いただければと存じます。
その後、再度レビュー依頼いただけますと幸いです。

何卒よろしくお願いいたします。

jodyyang007 commented:
- ログイン状態の場合でも、URLを直接入力して自身が出品していない売却済み商品の商品購入ページへ遷移しようとすると、トップページに遷移する動画ログインユーザー名と商品詳細ページの出品者名、soldoutの表記がわかるよう状態から購入画面への直接遷移の様子を撮影いただけますと幸いです。


>自身(jodyでログインしてます）が出品していない売却済み商品(商品名：最近、出品者の名前：最終課題、Item ID:11）
https://gyazo.com/db2dc27eb2c5c3d5de72619eed15b4dd
>URLを直接入力してトップページに遷移する
https://gyazo.com/014e0b30d0e0a3a5df87d6fb33b38587

上記のような動画でよろしいでしょうか。

- ログイン状態の場合でも、URLを直接入力して自身が出品した売却済み商品の商品情報編集ページへ遷移しようとすると、トップページに遷移する動画（商品情報編集機能実装時に未実装であった場合）
自身が出品した売却済み商品の「編集ページ」への遷移について撮影いただけますと幸いです。
>自身（最終課題でログインしてます）が出品した売却済み商品(商品名：最近、出品者の名前：最終課題、Item ID:11）
https://gyazo.com/6b94b47aa20f39fb5b292ad92b04b944

お手数をおかけしますが、上記の動画でよいかどうかご確認の程お願いいたします。

minachanchan commented on app/controllers/items_controller.rb:
```diff
@@ -4,11 +4,24 @@ class ItemsController < ApplicationController
+
+
+def show
+  @item = Item.find(params[:id])
```
```ruby
@item = Item.find(params[:id])
```
11行目と20行目はbeforeアクションでリファクタリングできていますので削除しましょう！

jodyyang007 replied:
重複部分を削除しました。

minachanchan commented on app/controllers/orders_controller.rb:
```diff
@@ -0,0 +1,67 @@
+
+  
+  def index
+    @item = Item.find(params[:item_id])
```
```ruby
@item = Item.find(params[:item_id])
```

現在、こちらの記述が複数ございます。
同じ記述をひとまとめにすることでコードをよりシンプルにすることができます。
before_actionを使ったリファクタリングに挑戦しましょう！

jodyyang007 replied:
こちらも削除した

minachanchan commented on app/controllers/orders_controller.rb:
```diff
@@ -0,0 +1,67 @@
+    
+
+    def pay_item
+    Payjp.api_key = "sk_test_5fe4735ea4c3dd01b6331e40" 
```
```ruby
Payjp.api_key = "sk_test_5fe4735ea4c3dd01b6331e40"
```
こちらは環境変数化し直接鍵情報を記載しない形式に変更しましょう！

jodyyang007 replied:
変更しました。

minachanchan commented on app/models/order_shipping.rb:
```diff
@@ -0,0 +1,23 @@
+
+  with_options presence: true do
+    validates :postal_code, format: { with: /\A[0-9]{3}-[0-9]{4}\z/, message: "is invalid. Include hyphen(-)" }
+    validates :prefecture_id, presence: { message: "can't be blank" }
```
```ruby
validates :prefecture_id, presence: { message: "can't be blank" }
```
「---」に該当するidを弾くバリデーションを指定しましょう！

jodyyang007 replied:
validates :prefecture_id, numericality: {other_than: 0, message: "can't be blank"}に変更した。
また、item.rbのカテゴリーに関するバリュエーションも変更した。

minachanchan commented on spec/factories/order_shippings.rb:
```diff
@@ -0,0 +1,12 @@
+    city { 'Shinjuku' }
+    street_address { '123' }
+    phone_number { '09012345678' }
+    item_id { 1 }
+    user_id { 1 }
```
```ruby
    item_id { 1 }
    user_id { 1 }
```

「user_id」と「item_id」は、order_shippings_spec.rbで生成しましょう。

外部キーカラムに、直接数値を設定する実装は推奨されていないためです。
「user_id」と「item_id」については、こちらのファイルではなく、○○_spec.rbで生成するように変更しましょう。
以下カリキュラムも合わせてご参考にいただけますと幸いです。
```
https://master.tech-camp.in/v2/curriculums/6771
Formオブジェクトパターンで実装した場合のテストコードの書き方
```

jodyyang007 replied:
テストコードに入れました。
 'item_id が空では保存できないこと'
'user_id が空で は保存できないこと'

minachanchan commented on spec/models/order_shipping_spec.rb:
```diff
@@ -0,0 +1,80 @@
+  describe '住所保存できる' do
+    context '住所保存できる場合' do
+      it "postal_codeとprefecture、city、street_address、phone_number,tokenが存在すれば保存できる" do
+        expect(@order_shipping).to be_valid
```
建物名の入力がなくても登録できることを確認するテストを追加しましょう！

jodyyang007 replied:
it "建物名が空でも保存できる" do
        @order_shipping.building_name = ''
        expect(@order_shipping).to be_valid
      end

minachanchan commented on spec/models/order_shipping_spec.rb:
```diff
@@ -0,0 +1,80 @@
+        @order_shipping.prefecture_id = nil
+        @order_shipping.valid?
+        expect(@order_shipping.errors.full_messages).to include("Prefecture can't be blank")
+      end
```
active_hashで実装した項目に関しては、「---」に対応するidが選択された場合保存できないことを確認するテストコードを実装しましょう！

jodyyang007 replied:
it "prefectureが「---」を選択した状態では保存できない" do
        @item.prefecture_id = Prefecture.find_by(name: '---').id 
        @item.valid?
        expect(@item.errors.full_messages).to include("Prefecture can't be blank")
      end

minachanchan commented on spec/models/order_shipping_spec.rb:
```diff
@@ -0,0 +1,80 @@
+        @order_shipping.phone_number = '090-12345678'
+        @order_shipping.valid?
+        expect(@order_shipping.errors.full_messages).to include('Phone number is invalid')
+      end
```
電話番号に関して以下を確認するテストをそれぞれ記述しましょう！

```
・9桁以下では購入できない
・12桁以上では購入できない
```

minachanchan reviewed (commented):
実装お疲れ様です！レビューのご依頼ありがとうございます。
キャプチャのご提示もお礼申し上げます！

コメントいたしましたので、ご確認頂けますと幸いでございます。


実装完了まであと少し、一緒に頑張りましょう！
修正の後、再度レビューのご依頼頂けますようお願い申し上げます。

jodyyang007 committed:
11行目と20行目　@item = Item.find(params[:id])削除

jodyyang007 committed:
重複@item = Item.find(params[:item_id])削除

jodyyang007 committed:
prefecture_idバリュエーション修正

validates :prefecture_id, presence: { message: "can't be blank" }
「---」に該当するidを弾くバリデーションを指定しましょう！

jodyyang007 committed:
「---」に該当するidを弾くバリデーションを指定

validates :prefecture_id, numericality: {other_than: 0, message: "can't be blank"}

jodyyang007 commented:
単体テストコード：
https://gyazo.com/acae2bf6e8c788a0235876d8a868268c

jodyyang007 committed:
単体テストコード及び環境変数

minachanchan commented on app/controllers/orders_controller.rb:
```diff
@@ -0,0 +1,63 @@
+      return redirect_to root_path
+    elsif @order_shipping.valid? && @order.save
+      @item.mark_as_sold_out
+      redirect_to root_path
```
現在の実装ですと、購入が正常にできてトップページに戻った際に、配送先住所情報テーブルにデータの保存がされないのではないかと存じます。
formオブジェクトクラスのsaveメソッドを呼び出し、配送先住所情報を保存できるよう実装しましょう！
もし、住所情報が正常に保存できている場合はご放念くださいませ。

jodyyang007 replied:
https://gyazo.com/dd513eee957dbe3083bc7baf53134588
住所は通常で保存されているようです。

jodyyang007 commented:
間違いまして、minachanchan さんにreview requestしてしまいました。すみません。

minachanchan commented on app/controllers/items_controller.rb:
```diff
@@ -4,11 +4,21 @@ class ItemsController < ApplicationController
+
 def edit
-  if current_user != @item.user
+   if current_user != @item.user || @item.sold_out?
```
```ruby
@item.sold_out?
```
こちらのsold_out?メソッドはどのような内容で定義されていますでしょうか？
購入履歴の有無を確認するように実装できているかご確認いただければと存じます！
もしこちらすでに実装済みでしたらご放念くださいませ。

jodyyang007 replied:
|| @item.sold_out?を削除しました。

jodyyang007 replied:
http://localhost:3000/items/id/edit
で直接URLを入れても編集できないと実装したいので、 || @item.sold_out?を追加した。

minachanchan commented on spec/models/order_shipping_spec.rb:
```diff
@@ -0,0 +1,94 @@
+        @order_shipping.phone_number = '090-12345678'
+        @order_shipping.valid?
+        expect(@order_shipping.errors.full_messages).to include('Phone number is invalid')
+      end
```
[こちら](https://github.com/jodyyang007/furima-39373/pull/19#discussion_r1330072339)について改めてご確認と実装をお願いいたします！

jodyyang007 replied:
「ー」があると保存できないとデストしたので、上記を記載しました。
・9桁以下では購入できない
・12桁以上では購入できない
追加しました
https://gyazo.com/b8dc18a99a455d5e03a5d45a371938df

minachanchan reviewed (commented):
実装お疲れ様です！レビューのご依頼ありがとうございます。

コメントいたしましたので、ご確認頂けますと幸いでございます。

実装完了まで引き続き、一緒に頑張りましょう！
ご不明点がございましたらお気軽にメンターまでご質問くださいませ！
修正の後、再度レビューのご依頼頂けますようお願い申し上げます。

jodyyang007 committed:
修正２

ShotaOtake commented on app/controllers/items_controller.rb:
```diff
@@ -4,11 +4,21 @@ class ItemsController < ApplicationController
+
 def edit
-  if current_user != @item.user
+   if current_user != @item.user || @item.sold_out?
```
◯下記の内容を修正しましょう！

https://github.com/jodyyang007/furima-39373/pull/19/files#r1332959556

【@item.order】が存在しているかどうかの条件で判定を行いましょう！

jodyyang007 replied:
def edit
   if current_user != @item.user|| @item.order.present? #|| @item.sold_out?
    redirect_to root_path
  else
  end
end
でよいでしょうか。

ShotaOtake reviewed (commented):
レビュー依頼を頂きありがとうございます！
追加で修正依頼のコメントをさせて頂きましたので、ご確認の上修正をお願い致します。

完成まであと一歩です！
引き続き実装頑張っていきましょう！

jodyyang007 committed:
|| @item.order.present?の修正

sh10n7 commented on app/controllers/orders_controller.rb:
```diff
@@ -0,0 +1,63 @@
+    private
+
+  def order_params
+    params.require(:order_shipping).permit(:postal_code, :prefecture_id, :city, :street_address, :building_name, :phone_number,:card_number, :expiry_form, :card_cvc).merge(item_id: @item.id, user_id: current_user.id, token: params[:token])
```
```
params.require(:order_shipping).permit(:postal_code, :prefecture_id, :city, :street_address, :building_name, :phone_number,:card_number, :expiry_form, :card_cvc).merge(item_id: @item.id, user_id: current_user.id, token: params[:token])
```
```:card_number, :expiry_form, :card_cvc```
上記のカード情報はデータベースには保存してはいけない情報となります。ストロングパラメーターから削除しましょう！

jodyyang007 replied:
:card_number, :expiry_form, :card_cvc削除しました

sh10n7 commented on app/models/order_shipping.rb:
```diff
@@ -0,0 +1,23 @@
+    validates :prefecture_id, numericality: {other_than: 0, message: "can't be blank"}
+    validates :city
+    validates :street_address
+    validates :phone_number, format: { with: /\A[0-9]{11}\z/ }
```
```
validates :phone_number, format: { with: /\A[0-9]{11}\z/ }
```
上記の記述ですと、11桁の電話番号しか登録できません。
実装条件にあるように10桁、または11桁の電話番号が保存できるように正規表現を見直しましょう！

jodyyang007 replied:
validates :phone_number, format: { with: /\A\d{10}\z|\A\d{11}\z/ }へ修正しました

sh10n7 commented on app/models/order_shipping.rb:
```diff
@@ -0,0 +1,23 @@
+  include ActiveModel::Model
+  attr_accessor :postal_code, :prefecture_id, :city, :street_address, :building_name, :phone_number, :item_id, :user_id
+  attr_accessor :token
+  validates :token, presence: true
```
```
validates :token, presence: true
```
presence: trueのバリデーションが複数箇所で使用されています。
上記の```validates :token```もwith_optionsのブロックに含んで、リファクタリングをしましょう。

jodyyang007 replied:
修正しました。

sh10n7 commented on spec/factories/order_shippings.rb:
```diff
@@ -0,0 +1,10 @@
+    street_address { '123' }
+    phone_number { '09012345678' }
+    token {"tok_abcdefghijk00000000000000000"}
+  end
```
建物のデータもダミーデータを作成しましょう！

jodyyang007 replied:
building_name { nil }にしました。

sh10n7 commented on spec/models/order_shipping_spec.rb:
```diff
@@ -0,0 +1,108 @@
+
+    context '住所保存できる場合' do
+      it "postal_codeとprefecture、city、street_address、phone_number、tokenが存在すれば保存できる" do
+        expect(@order_shipping).to be_valid
```
[こちら](https://github.com/jodyyang007/furima-39373/pull/19/files#r1330071418)が修正されていないようです。
ご確認の上、修正をお願いいたします。

sh10n7 reviewed (commented):
レビュー依頼ありがとうございます。
修正点確認させていただきました。

追加で修正いただきたい箇所をコメントしております。ご確認いただけますと幸いです。
​
修正頂いた後、コミットプッシュを確認いただき、再度レビュー依頼のご提出をお願い致します！

その際、**修正後のテスト実行結果の画像**も添付いただきますようお願いいたします。

ご不明点や疑問点ございましたら、チャット・コネクトメンターにご相談ください！
どうぞ、よろしくお願いいたします。

jodyyang007 committed:
:card_number, :expiry_form, :card_cvc削除

jodyyang007 committed:
validates :phone_number, format: { with: /\A\d{10}\z|\A\d{11}\z/ }

電話番号桁数修正

jodyyang007 committed:
validates :token, presence: trueの修正

jodyyang007 commented:
【@item.order】が存在しているかどうかの条件で判定の修正画像
https://gyazo.com/c178aba7b0c43c722dc82a6e8e8318ac
https://gyazo.com/25e89c98487928ee0442a920403c6e94

jodyyang007 committed:
修正3

jodyyang007 commented:
単体テストコード（単体テストコードは編集してませんので、同じの画像です）
https://gyazo.com/114a438852db47d673eadc96044b326a

sh10n7 commented on app/controllers/orders_controller.rb:
```diff
@@ -0,0 +1,63 @@
+    end
+
+
+    def show
+    if @item.sold_out? || current_user != @item.user
+        redirect_to root_path
+      elsif current_user != @item.user
+        redirect_to new_user_session_path
+      end
+    end
```
```
    def show
    if @item.sold_out? || current_user != @item.user
        redirect_to root_path
      elsif current_user != @item.user
        redirect_to new_user_session_path
      end
    end
```
これらは不要な記述となるため、削除しておきましょう。

jodyyang007 replied:
削除しました。

sh10n7 replied:
[こちら](https://github.com/jodyyang007/furima-39373/pull/19/files#r1334281315)の修正がお済みでないようです。
ご確認の上、修正をお願いいたします！

sh10n7 commented on app/models/order_shipping.rb:
```diff
@@ -0,0 +1,23 @@
+class OrderShipping
+  include ActiveModel::Model
+  attr_accessor :postal_code, :prefecture_id, :city, :street_address, :building_name, :phone_number, :item_id, :user_id
+  attr_accessor :token
```
```
  attr_accessor :postal_code, :prefecture_id, :city, :street_address, :building_name, :phone_number, :item_id, :user_id
  attr_accessor :token
```
```attr_accessor```の記述が複数回されているため、こちら1つにまとめましょう。

jodyyang007 replied:
修正しました。

sh10n7 reviewed (commented):
レビュー依頼ありがとうございます。
修正点確認させていただきました。

追加で修正いただきたい箇所をコメントさせていただきましたので、ご確認をお願いいたします。

また、修正後はコミットプッシュを確認いただき、再度**レビュー依頼フォーム**のご提出
をお願いいたします。

ご不明点や疑問点ございましたら、チャット・コネクトメンターにご相談ください！
よろしくお願いいたします。

jodyyang007 committed:
修正3

sh10n7 reviewed (commented):
レビュー依頼ありがとうございます。
修正点確認させていただきました。

追加で修正いただきたい箇所をコメントさせていただきましたので、ご確認をお願いいたします。

また、修正後はコミットプッシュを確認いただき、再度**レビュー依頼フォーム**のご提出
をお願いいたします。

ご不明点や疑問点ございましたら、チャット・コネクトメンターにご相談ください！
よろしくお願いいたします。

jodyyang007 committed:
showの削除

jodyyang007 committed:
showの削除、sold_outの削除

Kentozzz commented on app/models/order_shipping.rb:
```diff
@@ -0,0 +1,23 @@
+  include ActiveModel::Model
+  attr_accessor :postal_code, :prefecture_id, :city, :street_address, :building_name, :phone_number, :item_id, :user_id,:token
+
+  #belongs_to :order
```
```
#belongs_to :order
```
こちらのアソシエーションは不要でしたら削除しましょう。

jodyyang007 replied:
削除しました。

Kentozzz commented on db/migrate/20230916061328_create_addresses.rb:
```diff
@@ -0,0 +1,14 @@
+      t.string :postal_code, null: false
+      t.string :city, null: false
+      t.string :street_address, null: false
+      #t.string :building_name, null: false
```
```
#t.string :building_name, null: false
```
こちらのコメントアウトは削除しましょう。

jodyyang007 replied:
削除しました。

Kentozzz commented on db/migrate/20230918040918_add_sold_out_to_items.rb:
```diff
@@ -0,0 +1,5 @@
+class AddSoldOutToItems < ActiveRecord::Migration[6.0]
+  def change
+    add_column :items, :sold_out, :boolean
```
```
add_column :items, :sold_out, :boolean
```
READMEと異なる実装をおこなっていらっしゃるようですので、READMEの修正もおこなっておきましょう。

jodyyang007 replied:
README修正

Kentozzz commented on app/views/orders/index.html.erb:
```diff
@@ -41,8 +42,10 @@
+          <div id="number-form" class="input-default"></div>
         </div>
-        <%= f.text_field :hoge, class:"input-default", id:"card-number", placeholder:"カード番号(半角英数字)", maxlength:"16" %>
+        <%#= f.text_field :number, class:"input-default", id:"card-number", placeholder:"カード番号", maxlength:"16" %>
```
```
<%#= f.text_field :number, class:"input-default", id:"card-number", placeholder:"カード番号", maxlength:"16" %>
```
不要なコメントアウトは削除しましょう。
残したい場合は「残す理由」も添えてコメントアウトしておきましょう。
また、レビュー時はコメントを添えるだけで問題ないですが他者に見せる前に削除していただけると良いかと存じます。

※こちら以外にも残っておりますので、合わせて削除するか追加の説明コメントを入れておきましょう。

jodyyang007 replied:
削除しました。

Kentozzz reviewed (commented):
レビュー依頼ありがとうございます！
コメントしましたので、ご確認をお願いいたします！

jodyyang007 committed:
修正４

Kentozzz reviewed (commented):
レビュー依頼ありがとうございます！
LGTMとさせていただきます！実装お疲れ様でした！
このブランチをマージしmainブランチでプルを行いましょう！
内容が反映されていることを確認して次の実装に進みましょう！
